lnwallet/btcwallet/psbt.go

```diff
@@ -85,6 +87,7 @@ func (b *BtcWallet) FundPsbt(packet *psbt.Packet, minConfs int32,
 	return b.wallet.FundPsbt(
 		packet, keyScope, minConfs, accountNum, feeSatPerKB,
 		b.cfg.CoinSelectionStrategy,
+		wallet.WithCustomChangeScope(changeScope),
```

Comment:
Similar to the btcwallet PR, passing nil here as the 'custom' scope still doesn't sit completely right with me,

Reply:
I would also prefer a default value here, otherwise I had to dig deep into the btcwallet lib to find that its defaulted to p2tr, so I think setting it here, makes things more clearer ?

Reply:
ok, I will pass the `customscope` as an argument only if it's not nil !

> I would also prefer a default value here, otherwise I had to dig deep into the btcwallet lib to find that its defaulted to > p2tr, so I think setting it here, makes things more clearer ?

I am not really sure about that. The logic of taking P2TR by default is already in btcwallet so there will be some duplication of code?

The `godoc` of `b.wallet.FundPsbt` explains the default behaviour. Maybe it is not well explained, and I need to explain better the behaviour (something similar as in [walletkit.proto?)](https://github.com/lightningnetwork/lnd/pull/7348/files/8744e2cd8231ee0af1b7de6e20f43f7898b9fea6..c8c49f010b3d759dea3404a413146c7f08fd3a5b#diff-217f3c5bea0446bf0241e3f9323fd97cfa73ef259de31b4926cb1e6aa5f6f0a6R975)

Reply:
I understand where you are coming from @ziggie1984 for the default accounts in lnd. But there are also user-defined accounts that can have any key scope, and for those the default change is not p2tr but the input key scope.

Now that we are talking about that - how does this code behave when a different change scope is selected for a user-defined accounts that doesn't exist for that change scope?

Reply:
Actually, if the user define a change scope that doesn't exist for this account, we will return an error"
https://github.com/btcsuite/btcwallet/blob/master/waddrmgr/manager.go#L627

> (ties into https://github.com/lightningnetwork/lnd/issues/7265?)

However, you got a point. Currently, the behaviour we mentioned in this issue shouldn't exist (apparently). So if the user (even if he shouldn't), has an account with 2 different scopes, then we will select one randomly and, depending on which one is selected, we will return an error (in FetchScopedKeyManager)

So we definitely either:

- Don't allow to create 2 scopes for the same user account
- Allow it but make "LookUpAccount()" deterministic

Reply:
For option 1, there is still a problem for existing user accounts that have been created for multiple key scopes. So perhaps both need to be done?

I think the alternative is to add a `account_key_scope` field to `FundPsbt` to make input selection explicit?

Reply:
Or should the behavior for inspect selection for user accounts be the same as for default? Select inputs from all key scopes?

Reply:
> So perhaps both need to be done?

Yeah indeed, I think you are right.
For me, for the option 2, the `account_key_scope` is definitely a good idea (and will be consistent with the possibility to choose a key scope for the change

I will update https://github.com/lightningnetwork/lnd/issues/7265 to mention this

Reply:
> Do you think that it's not enough?

Yes I think you made it clear in the comment now, thx

Reply:
will the account_key_scope default to something when not set, or means explicit it has to be set ?

Will we allow 2 scopes for one account, if so I find it good to behave like the default account and select funds from all scopes ?

Reply:
> Will we allow 2 scopes for one account

I don't think this is a desired  behaviour:
https://github.com/lightningnetwork/lnd/issues/7265

Reply:
Regarding the first concern:
> Similar to the btcwallet PR, passing nil here as the 'custom' scope still doesn't sit completely right with me

Now custom scope can't be nil

Reply:
The main problem (not the only one) @joostjager and I have found i concerns the default behaviour we want for user account with more than 1 key scope.
Indeed, the change output keyscope has to be among the keyscopes of the user account.
But in the specific case where this account doesn't have the P2WKH keyscope, how should we default?

In any case, the multi-keyscope user account currently seems to be a problem and needs to be solve first:
https://github.com/lightningnetwork/lnd/pull/7348

I will resolve this thread for now, but feel free to unresolve or open a new thread to continue discussion!